src/Libraries/CoreNodeModelsWpf/Charts/Utilities/Export.cs

```
@@ -4,37 +4,62 @@
+                        skChart = new SKCartesianChart(cartesianChart) { Width = img_width, Height = img_height, };
+                        break;
+                    case PieChart pieChart:
+                        skChart = new SKPieChart(pieChart) { Width = img_width, Height = img_height, };
+                        break;
+                    default:
+                        skChart = new SKGeoMap(gcontrol) { Width = img_width, Height = img_height, };
+                        break;
+                }
 
-            SaveFileDialog dialog = new SaveFileDialog();
-            dialog.FileName = "NodeModelChart"; // Default file name
-            dialog.DefaultExt = ".png"; // Default file extension
-            dialog.Filter = "Image files (*.png) | *.png"; // Filter files by extension
+                SaveFileDialog dialog = new SaveFileDialog();
+                dialog.FileName = "NodeModelChart"; // Default file name
+                dialog.DefaultExt = ".png"; // Default file extension
+                dialog.Filter = "Image files (*.png) | *.png"; // Filter files by extension
 
-            // Show save file dialog box
-            Nullable<bool> result = dialog.ShowDialog();
+                // Show save file dialog box
+                bool? result = dialog.ShowDialog();
 
-            // Process save file dialog box results
-            if (result == true)
+                // Process save file dialog box results
+                if (result == true)
+                {
+                    // Save document
+                    string filename = dialog.FileName;
+                    skChart.SaveImage(filename, SkiaSharp.SKEncodedImageFormat.Png);
+                }
+            }
+            catch (Exception ex)
             {
-                // Save document
-                string filename = dialog.FileName;
-                image.Save(filename, ImageFormat.Png);
+                Console.WriteLine("Chart Image Export Error:" + ex.Message);
```

Do we need this try-catch or was this also for your debugging?

---

Is Console.WriteLine visible in the Dynamo console UI ? or at least in the console logs ?
Do we have access to the Dynamo Logger here ?
I say we should let the exception bubble up, unless it is something very specific you are trying to catch here @zeusongit .

---

This was for test as well, LC2 has summed up all charts into 2 types, SKCartesian and Pie, but there is one more Geo charts which are not yet implemented, so was testing if any of the existing charts throw an exception while exporting image.